aea/connections/local/connection.py

```diff
@@ -178,9 +191,10 @@ def search_agents(self, public_key: str, search_id: int, query: Query) -> None:
         if query.model is None:
             result = list(set(self.services.keys()))
         else:
-            for agent_public_key, description in self.agents.items():
-                if query.model == description.data_model:
-                    result.append(agent_public_key)
+            for agent_public_key, descriptions in self.agents.items():
+                for description in descriptions:
+                    if query.model == description.data_model:
+                        result.append(agent_public_key)
```

Comment:
notice, this was the old version. Here we should call the `check` method of the `query` object, which is now implemented.

Reply:
it is based on the service functionality.

Reply:
Oh ok, I think we can remove also `search_agents` since it is not used (right?)